proposals/extensions.md

```diff
@@ -715,6 +956,30 @@ The search proceeds as follows:
 - If no extension indexer is found to be suitable for the element access
   in any enclosing scope, a compile-time error occurs.
 
+#### This access
+
+We modify the [this access rules](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#12813-this-access) as follows:
+
+A this_access has one of the following meanings:
+- [...]
+- ***When this is used in a primary_expression within an instance method or instance accessor of an extension with a class underlying type,
```

Comment:
> with a class underlying type

A class underlying type, or an underlying type known to be a reference type? For example, an interface type is not a class and is not a struct, and is not a type parameter.